### PR TITLE
build(devs-infra): migrate to Yarn 2

### PR DESCRIPTION
**Pull request by ahnpnl**
## Summary

- Migrate repo to Yarn 2
- Adjust CI to use Yarn 2 (following CI config from Jest repo)

## Test plan

Green CI

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
**N.A.**

### CODE REVIEW

File: e2e/test-app-v9/.gitignore [hunk 1]
```diff
@@ -41,3 +41,10 @@ src/__tests__
 # System Files
 .DS_Store
 Thumbs.db
+.yarn/*
+!.yarn/cache
+!.yarn/patches
+!.yarn/plugins
+!.yarn/releases
+!.yarn/sdks
+!.yarn/versions
```
Comment (thymikee) [inline e2e/test-app-v9/.gitignore:50]: why are these excluded?

Comment (ahnpnl) [reply e2e/test-app-v9/.gitignore:50]: I followed the instruction from Yarn 2 documentation so I wasn't sure ><

Comment (ahnpnl) [reply e2e/test-app-v9/.gitignore:50]: Btw, i see Jest repo excludes some of the ones above too

Comment (thymikee) [reply e2e/test-app-v9/.gitignore:50]: Can you point me to the docs stating that?

Comment (thymikee) [reply e2e/test-app-v9/.gitignore:50]: Ok, found https://yarnpkg.com/getting-started/qa/#which-files-should-be-gitignored. I'm good then

**Pull request merged**